web/src/components/Editor/hooks/useEditorValidate.ts

```diff
@@ -0,0 +1,28 @@
+import {useCallback} from 'react';
+import {SupportedEditors} from 'constants/Editor.constants';
+import {expressionQLang} from '../Expression/grammar';
+import {interpolationQLang} from '../Interpolation/grammar';
+import {selectorQLang} from '../Selector/grammar';
+
+const langMap = {
+  [SupportedEditors.Expression]: expressionQLang,
+  [SupportedEditors.Interpolation]: interpolationQLang,
+  [SupportedEditors.Selector]: selectorQLang,
+} as const;
+
+const useEditorValidate = () => {
```

Comment:
Probably it's not needed as a `hook`. It could be a simple function that you can import and use it. It does not have to be created as part of the render flow.

Reply:
Fixing this in the follow pr